src/OrchardCore/OrchardCore.FileStorage.FileSystem/FileSystemStore.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OrchardCore.FileStorage.FileSystem
+{
+    public class FileSystemStore : IFileStore
+    {
+        private readonly string _fileSystemPath;
+
+        public FileSystemStore(string fileSystemPath)
+        {
+            _fileSystemPath = Path.GetFullPath(fileSystemPath);
+        }
+
+        public Task<IFileStoreEntry> GetFileInfoAsync(string path)
+        {
+            var physicalPath = GetPhysicalPath(path);
+
+            var fileInfo = new FileInfo(physicalPath);
```

Comment:
I wonder if it’s better to new up a PhsicalFileInfo here rather than a generic one

Reply:
Makes sense to me. I'll give it a try.